modules/flann/include/opencv2/flann/heap.h

```diff
@@ -49,117 +54,189 @@ namespace cvflann
 template <typename T>
 class Heap
 {
-
     /**
      * Storage array for the heap.
      * Type T must be comparable.
      */
     std::vector<T> heap;
-    int length;
 
+    struct greater_equal
+    {
+        bool operator()(const T& x, const T& y) const
+        {
+            return y < x;
+        }
+    };
+
+public:
     /**
-     * Number of element in the heap
+     * \brief Constructs a heap with a pre-allocated capacity
+     *
+     * \param capacity heap maximum capacity
      */
-    int count;
-
-
+    Heap(const int capacity)
+    {
+        reserve(capacity);
+    }
 
-public:
     /**
-     * Constructor.
+     * \brief Move-constructs a heap from an external vector
      *
-     * Params:
-     *     sz = heap size
+     * \param vec external vector
      */
+    Heap(std::vector<T>&& vec)
+        : heap(std::move(vec))
+    {
+        static greater_equal comp;
+        std::make_heap(heap.begin(), heap.end(), comp);
```

Comment:
Lets avoid using of static local declarations. Static code analyzers dislikes that.

This code should work:
```
std::make_heap(heap.begin(), heap.end(), greater_equal());
```

or even with `std::greater<T>()`